pkg/genlib/generator_interface.go

```diff
@@ -48,8 +48,8 @@ const (
 	FieldTypeFlattened       = "flattened"
 	FieldTypeGeoPoint        = "geo_point"
 
-	FieldTypeTimeRange  = 3600 // seconds
-	FieldTypeTimeLayout = "2006-01-02T15:04:05.999999Z07:00"
+	FieldTypeDurationSpan = 1000 // milliseconds
```

Comment:
can we use typed constants for these to make it clearer e.g. `1000 * time.Millisecond` ?

Reply:
ah - rereading the PR description i understand this better. 

instead of spreading out across an hour, each event is between 0 and 1 s after the previous event. 

this could have a really significant impact on the time span over which events are generated... what's the reason for making this change?

Reply:
> can we use typed constants for these to make it clearer e.g. `1000 * time.Millisecond` ?

I use the constant as param of `rand.Intn`, I cannot use a duration unluckily

Reply:
> this could have a really significant impact on the time span over which events are generated... what's the reason for making this change?

yes, it has a significant impact. but the fields for this difference will apply should not be significant.
I mean that if you want a precise time span, you should you `period`. if you are not using period I wanted to achieve to things:
1. no floating back and forth in time the generated dates (always progressive from now)
2. keeping the time span shorter.

I'm open the reconsider my goals :)
either both or one of them only :)

I'm trying to figure out what would be the most "desired" default behavior

Reply:
on the first part, we can just cast the `Duration` to an `int` in order to pass it to `Intn`?

on the second, sure that sounds fine, but it would be great to see design decisions like this recorded somewhere :)

Reply:
> it would be great to see design decisions like this recorded somewhere

well. I hoped to have recorded it in the PR description :D

Reply:
> on the first part, we can just cast the `Duration` to an `int` in order to pass it to `Intn`?

but then we could end up with nanoseconds spans because it will be `Intn` of nanoseconds. since we print date with milliseconds resolution we will end up with date with no span

Reply:
but `Duration`s are already in nanoseconds. we are just scaling up to ms before doing the random selection... i think?

Reply:
> but `Duration`s are already in nanoseconds. we are just scaling up to ms before doing the random selection... i think?

```
 time.Duration(customRand.Intn(1000)) * time.Millisecond
```

is different from
```
 time.Duration(customRand.Intn(1000000000))
```

the first generates from 0 to 1000 milliseconds, ie, it has a duration resolution of milliseconds

the latter generates from 0 to 1000000000 nanoseconds, with a nanoseconds duration resolution that we finally lost when we print at milliseconds resolution :)

Reply:
> the two examples produce equivalent durations with different precisions, right?

exactly, that's the point

`customRand.Intn(1000) * time.Millisecond` will return `event1: 15:10:22.058`, `event2: 15:10:22.189`.
while the second will return `event1: 15:10:22.0580000110`, `event2: 15:10:22.0580024090`.

since when we print the `time.Time` we print milliseconds resolution only, in the second case we end up with the same time